wave_utils/src/math.cpp

```diff
@@ -3,343 +3,87 @@
 
 namespace wave {
 
-int sign(double x) {
-  return (x < 0) ? -1 : 1;
+int randi(int ub, int lb) {
+    return rand() % lb + ub;
 }
 
-float deg2rad(float d) {
-  return d * (M_PI / 180);
+double randf(double ub, double lb) {
+    double f = (double) rand() / RAND_MAX;
+    return lb + f * (ub - lb);
 }
 
-float rad2deg(float r) {
-  return r * (180 / M_PI);
-}
-
-int fltcmp(float f1, float f2) {
-  if (fabs(f1 - f2) <= 0.0001) {
-    return 0;
-  } else if (f1 > f2) {
-    return 1;
-  } else {
-    return -1;
-  }
-}
-
-MatX kronecker_product(MatX A, MatX B) {
-  MatX C;
-  double a;
-
-  // setup
-  C.resize((A.rows() * B.rows()), (A.cols() * B.cols()));
-
-  // calculate kronecker product
-  for (int i = 0; i < A.rows(); i++) {
-    for (int j = 0; j < A.cols(); j++) {
-      a = A(i, j);
-      C.block(i * B.rows(), j * B.cols(), B.rows(), B.cols()) = a * B;
+int fltcmp(double f1, double f2) {
+    if (fabs(f1 - f2) <= 0.0001) {
```

Comment:
Fixed magic by introducing third param to include an acceptable float comparison threshold.

`fltcmp(double f1, double f2, double threshold=0.0001)`

with a default threshold of 0.0001